packages/grid/x-data-grid/src/models/gridSlotsComponentsProps.ts

```diff
@@ -1,28 +1,83 @@
+import * as React from 'react';
+import { CheckboxProps } from '@mui/material/Checkbox';
+import { TextFieldProps } from '@mui/material/TextField';
+import { FormControlProps } from '@mui/material/FormControl';
+import { SelectProps } from '@mui/material/Select';
+import { SwitchProps } from '@mui/material/Switch';
+import { ButtonProps } from '@mui/material/Button';
+import { IconButtonProps } from '@mui/material/IconButton';
+import { TooltipProps } from '@mui/material/Tooltip';
+import { PopperProps } from '@mui/material/Popper';
+import { TablePaginationProps } from '@mui/material/TablePagination';
+import { GridToolbarProps } from '../components/toolbar/GridToolbar';
+import { ColumnHeaderFilterIconButtonProps } from '../components/columnHeaders/GridColumnHeaderFilterIconButton';
+import { GridColumnMenuProps } from '../components/menu/columnMenu/GridColumnMenuProps';
+import { GridColumnsPanelProps } from '../components/panel/GridColumnsPanel';
+import { GridFilterPanelProps } from '../components/panel/filterPanel/GridFilterPanel';
+import { GridFooterContainerProps } from '../components/containers/GridFooterContainer';
+import { GridOverlayProps } from '../components/containers/GridOverlay';
+import { GridPanelProps } from '../components/panel/GridPanel';
+import { GridRowProps } from '../components/GridRowProps';
+import { GridCellProps } from '../components/cell/GridCellProps';
+
+// Overrides for module augmentation
+export interface BaseCheckboxPropsOverrides {}
+export interface BaseTextFieldPropsOverrides {}
+export interface BaseFormControlPropsOverrides {}
+export interface BaseSelectPropsOverrides {}
+export interface BaseSwitchPropsOverrides {}
+export interface BaseButtonPropsOverrides {}
+export interface BaseIconButtonPropsOverrides {}
+export interface BaseTooltipPropsOverrides {}
+export interface BasePopperPropsOverrides {}
+export interface CellPropsOverrides {}
+export interface ToolbarPropsOverrides {}
+export interface ColumnHeaderFilterIconButtonPropsOverrides {}
+export interface ColumnMenuPropsOverrides {}
+export interface ColumnsPanelPropsOverrides {}
+export interface FilterPanelPropsOverrides {}
+export interface FooterPropsOverrides {}
+export interface PaginationPropsOverrides {}
+export interface LoadingOverlayPropsOverrides {}
+export interface NoResultsOverlayPropsOverrides {}
+export interface NoRowsOverlayPropsOverrides {}
+export interface PanelPropsOverrides {}
+export interface PreferencesPanelPropsOverrides {}
+export interface RowPropsOverrides {}
+
+type SlotProps<Props, Overrides> = Partial<Props> & Overrides;
```

Comment:
We could make it in a more backward-compatible manner to make the props augmentation optional:

```suggestion
type SlotProps<Props, Overrides> = Partial<Props> & Overrides & Record<PropertyKey, any>;
```

### Examples
- works without the augmentation:
	```tsx
	<DataGrid
	  slotProps={{
	    footer: { status: 'connected' }
	  }}
	/>
	```
- works with augmentation:
	```tsx
	declare module '@mui/x-data-grid' {
	  interface FooterPropsOverrides {
	    status: 'connected' | 'disconnected';
	  }
	}
	
	<DataGrid
	  slotProps={{
	    footer: { status: 'connected' }
	  }}
	/>
	```

### Pros

- backward compatible - any custom props will work as before, no changes required

### Cons

- without props augmentation, TS won't complain if the wrong type is passed:
	```tsx
	<DataGrid
	  slotProps={{
	    footer: { status: "whatever, TS won't complain anyway" }
	  }}
	/>
	```
  Also, you won't get autocomplete suggestions for the custom props.
	
What do you think?
Tagging @mui/x if you want to share your opinion 🙂

Reply:
Thanks for the idea, IMO though, after adding backwards compatibility with `Record<PropertyKey, any>`, augmentation might not be that useful anymore.

If I think of myself as a developer, I'll be less likely to use augmentation when there's an easier workaround. Other way around, I'll be forced to augment my extra props but as a result, will avoid the cons you mentioned and enjoy a more intuitive `dx`.

Do you think it's worth it to add a small migration cost to give a slightly better dx? 

Thinking more on this, should doing either of augmentation and backward compatibility makes more sense to keep API clean/simple 🤔

Reply:
IMHO, I'd be in favor of having "restricted freedom" for the sake of DX sanity. It does seem that having the option to define/add additional props via type overrides is sufficient extendibility. 🤔

Reply:
In my mind the idea of adding `Record<PropertyKey, any>` was for the sake of easier migration. But eventually, we would have to remove it anyway (in v7).

So +1 to not proceed with it 👍